core/src/mast/serialization/mod.rs

```diff
@@ -232,35 +232,123 @@ impl Deserializable for MastForest {
             mast_forest
         };
 
+        let basic_block_decorators: Vec<(usize, DecoratorList)> =
+            read_block_decorators(source, &mast_forest)?;
+        for (node_id, decorator_list) in basic_block_decorators {
+            let node_id = MastNodeId::from_usize_safe(node_id, &mast_forest)?;
+
+            match &mut mast_forest[node_id] {
+                MastNode::Block(basic_block) => {
+                    basic_block.set_decorators(decorator_list);
+                },
+                other => {
+                    return Err(DeserializationError::InvalidValue(format!(
+                        "Expected mast node with id {node_id} to be a basic block, found {other:?}"
+                    )))
```

Comment:
```suggestion
                    return Err(DeserializationError::InvalidValue(format!(
                        "expected mast node with id {node_id} to be a basic block, found {other:?}"
                    )))
```

Supernit: Error messages should be lowercase so they follow a common style.
> The error message given by the Display representation of an error type should be lowercase without trailing punctuation, and typically concise.
> https://rust-lang.github.io/api-guidelines/interoperability.html